### PR TITLE
Change deprecated ggplot2 syntax to current recommendations

### PR DESCRIPTION
**Pull request by bwiernik**
#### Summary:

*ggplot2* recently added a `linewidth` argument to replace `size` for line geoms and elements. Currently, loading *rstan* or a package that loads it produces a deprecation warning about the `size` argument. This PR replaces `size` with `linewidth` for line geoms and elements in the *rstan* plotting functions. The PR also updates a few other instances of deprecated *ggplot2* syntax with the recommended alternatives (replacing `aes_string()` with `aes(.data$...)`, replacing `"..density.."` with `after_stat(density)`).

#### Intended Effect:

No visual changes to plot output. Removal of deprecation warnings on package load.

#### How to Verify:

Output is unchanged for the following plots with the current `develop` version and this PR:

```r
library(brms)
library(rstan)
m <- brm(mpg ~ disp, data = mtcars)
stan_dens(m$fit)
stan_hist(m$fit)
stan_plot(m$fit)
```

#### Documentation:

https://github.com/tidyverse/ggplot2/blob/main/NEWS.md

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Brenton M. Wiernik

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: GPLv3 (http://opensource.org/licenses/GPL-3.0)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: rstan/rstan/R/stan_plot.R [hunk 1]
```diff
@@ -10,34 +10,34 @@ stan_trace <- function(object, pars, include = TRUE,
...
+
   thm <- rstanvis_theme()
   clrs <- rep_len(rstanvis_aes_ops("chain_colors"), plot_data$nchains)
-  
+
   base <-
     ggplot2::ggplot(
       plot_data$samp,
-      ggplot2::aes_string(x = "iteration", y = "value", color = "chain")
+      ggplot2::aes(x = .data$iteration, y = .data$value, color = .data$chain)
```
Comment (hsbadr) [inline rstan/rstan/R/stan_plot.R]: I think there's no need for `.data$` with `aes()`. This applies everywhere.
```suggestion
      ggplot2::aes(x = iteration, y = value, color = chain)
```

**Comment by hsbadr** [comment]
@bwiernik Can you address the comment above and rebase this on the `experimental` branch?

File: rstan/rstan/R/stan_plot_helpers.R [hunk 2]
```diff
@@ -536,20 +536,20 @@ color_vector_chain <- function(n) {
...
+  base <- ggplot2::ggplot(df, ggplot2::aes(x = .data$x, y = .data$y)) + xy_labs
   graph <- base + ggplot2::geom_violin(color = .NUTS_CLR, fill = .NUTS_FILL)
   if (chain == 0) return(graph)
   chain_clr <- color_vector_chain(ncol(df_x))[chain]
   chain_fill <- chain_clr
   chain_data <- data.frame(x = as.factor(df_x[, chain]), y = df_y[, chain])
   graph + ggplot2::geom_violin(
     data = chain_data,
-    mapping = ggplot2::aes_string("x", "y"),
+    mapping = ggplot2::ggplot2::aes(x = .data$x, y = .data$y),
```
Comment (hsbadr) [inline rstan/rstan/R/stan_plot_helpers.R]:
```suggestion
    mapping = ggplot2::aes(x = .data$x, y = .data$y),
```

**Review by bgoodri — COMMENTED** [review]
I am guessing this is OK, although I batched a few suggestions.

File: rstan/rstan/R/stan_plot.R (hunk 1 above)
Comment (bwiernik) [reply rstan/rstan/R/stan_plot.R]: No, the `.data$` is necessary to avoid global variable name NOTEs in CRAN checks

**Comment by bwiernik** [comment]
I removed another `aes_q()` that had been added on the experimental branch. The `.data` pronoun inside `aes()` is necessary because `aes(x = value)` will trigger a Undeclared Global Variable Name NOTE in R CMD CHECK. `data$value` or `data[['value']]` is the current recommended syntax for situations like this.

From `?aes_string`
> All these functions are soft-deprecated. Please use tidy evaluation idioms instead. Regarding `aes_string()`, you can replace it with `.data` pronoun. For example, the following code can achieve the same mapping as `aes_string(x_var, y_var)`.

```
x_var <- "foo"
y_var <- "bar"
aes(.data[[x_var]], .data[[y_var]])
```

See vignette("ggplot2-in-packages").

**Comment by jgabry** [comment]
Thanks @bwiernik!

**Comment by bgoodri** [comment]
Is this good to merge now?

**Comment by jgabry** [comment]
Were the linewidth changes merged already at some point? I don’t see them in the Files Changed view, but they are in the commit history. The Files Changed view now just shows aes and .data stuff, which looks good to go.

**Comment by bwiernik** [comment]
It looks like all of the commits but the most recent are showing as already merged.

**Comment by hsbadr** [comment]
Yes, I tested this in the `experimental`.

**Review by hsbadr — APPROVED** [review]

**Pull request merged**